File: ui.js
var params = {}
keys = ['latency', 'jitter', 'loss', 'rate', 'P', 'E_B', 'rho', 'P_isol', 'E_GB']

function load_params() {
    var request = new XMLHttpRequest();
    request.onreadystatechange = function() {
        if (this.readyState == 4 && this.status == 200) {
            params = JSON.parse(this.responseText);
            document.getElementById("dist").value = params.dist;
            document.getElementById("loss_model").value = params.loss_model;
            document.getElementById("ip").innerHTML = params.ip;
            for (i=0; i<keys.length; i++)
            {
                document.getElementById(keys[i]).setAttribute("value", params[keys[i]]);
            }
            loss_model_changed();
        }
    };
    request.open("GET", "/params", true);
    request.send();
}

function post_params() {
    
    params.dist = document.getElementById("dist").value;
    params.loss_model = document.getElementById("loss_model").value;
    for (i=0; i<keys.length; i++)
    {
        params[keys[i]] = document.getElementById(keys[i]).value;
    }
    var request = new XMLHttpRequest();
    request.open("POST", "/params", true);
    request.send(JSON.stringify(params));
}

function loss_model_changed() {
    if (document.getElementById("loss_model").value == 'random') {
        document.getElementById("loss-random-options").setAttribute("style", "display: contents;")
        document.getElementById("loss-gi-options").setAttribute("style", "display: none;")
    }
    else if (document.getElementById("loss_model").value == 'gi') {
        document.getElementById("loss-random-options").setAttribute("style", "display: none;")
        document.getElementById("loss-gi-options").setAttribute("style", "display: contents;")
    }
}

load_params();

File: ui.py
from core import get_my_ip, update_emulation_engine, init_iptables
from flask import Flask, send_from_directory, request
import json

app = Flask(__name__)

params = {
    'ip': get_my_ip(),
    'latency': '0',
    'jitter': '0',
    'dist': 'normal',
    'rate': '',
    'loss_model': 'random',
    'loss': '0',
    'P': 0, 'E_B': 10, 'rho': 0.99, 'P_isol': 0, 'E_GB': 60,
}

@app.route('/')
def index():
    return send_from_directory('.', 'ui.html')

@app.route('/<path:path>')
def static_(path):
    return send_from_directory('.', path)

@app.route('/params', methods=['GET','POST'])
def params_():
    global params
    if request.method == 'GET':
        return params
    else:
        params = json.loads(request.data)
        update_emulation_engine(params)
        return 'Good'

if __name__ == '__main__':
    init_iptables()
    print(f'\nEmulator IP: {params["ip"]}\nDashboard: http://{params["ip"]}:90/\n')
    app.run(host="0.0.0.0", port=90)
    